peer.go

```diff
@@ -1260,10 +1267,20 @@ func (p *peer) PingTime() int64 {
 func (p *peer) queueMsg(msg lnwire.Message, errChan chan error) {
 	select {
 	case p.outgoingQueue <- outgoingMsg{msg, errChan}:
+		return
 	case <-p.quit:
 		peerLog.Tracef("Peer shutting down, could not enqueue msg.")
-		if errChan != nil {
-			errChan <- fmt.Errorf("peer shutting down")
+	}
+
+	// If the caller is expecting an error, attempt to alert that the peer
+	// is shutting down.
+	if errChan != nil {
+		// NOTE: A non-blocking select is used in case this isn't the
+		// only error that gets signaled, and errors are not pulled
+		// from the other side of the errChan.
+		select {
```

Comment:
I don't think this `select` is is necessary. Observation is that once we are able to enqueue the message above, we'll return and we have handed over the responsibilty for the `errChan` to the reader from `p.outgoingQueue`. Since the `errChan` is buffered, it should never block on this send in case that hand off is not successful.

Reply:
agreed, I was erroring on the side of caution, but i will revert this